base/dcp_client_metadata.go

```diff
@@ -81,45 +93,47 @@ func NewDCPMetadataMem(numVbuckets uint16) *DCPMetadataMem {
 	return m
 }
 
-// Rollback resets the metadata, preserving EndSeqNo
-func (m *DCPMetadataMem) Rollback(ctx context.Context, vbID uint16) {
-	endSeqNo := m.metadata[vbID].EndSeqNo
-	m.metadata[vbID] = DCPMetadata{
-		VbUUID:          0,
-		StartSeqNo:      0,
-		EndSeqNo:        endSeqNo,
-		SnapStartSeqNo:  0,
-		SnapEndSeqNo:    0,
-		FailoverEntries: make([]gocbcore.FailoverEntry, 0),
+// Rollback resets vBucket metadata to the vBucket UUID and sequence number provided
+func (m *dcpMetadataBase) Rollback(ctx context.Context, vbID uint16, startSeqNo gocbcore.SeqNo) {
+	var rollbackVbuuid gocbcore.VbUUID
+	for _, failoverLog := range m.metadata[vbID].FailoverEntries {
+		if failoverLog.SeqNo <= startSeqNo {
+			rollbackVbuuid = failoverLog.VbUUID
+			break
+		}
 	}
-	TracefCtx(ctx, KeyDCP, "rolling back vb:%d with metadata set to %v", vbID, m.metadata[vbID])
+	m.metadata[vbID].VbUUID = rollbackVbuuid
+	m.metadata[vbID].StartSeqNo = startSeqNo
+	m.metadata[vbID].SnapStartSeqNo = startSeqNo
+	m.metadata[vbID].SnapEndSeqNo = startSeqNo
```

Comment:
We're no longer setting endSeqNo here - that seems like it would still be needed for a one-shot feed?

Reply:
We are just keeping the values we used to have here.